packages/dd-trace/src/telemetry/dependencies.js

```diff
@@ -71,7 +71,17 @@ function start (_config, _application, _host) {
 }
 
 function isDependency (filename, request) {
-  return request.indexOf(`.${path.sep}`) !== 0 && request.indexOf(path.sep) !== 0
+  const result = isDependencyWithSeparator(filename, request, '/')
+  if (result && path.sep !== '/') {
+    return isDependencyWithSeparator(filename, request, path.sep)
+  }
+  return result
```

Comment:
I'm not sure I understand this logic: 

First we call `isDependencyWithSeparator(filename, request, '/')` and then we call it again with `path.sep` if `path.sep` isn't `/`? 

Couldn't we just return `isDependencyWithSeparator(filename, request, path.sep)` ?

Reply:
In windows, you can use `/` or `\` in the requires. `require('../file')` is accepted, and `require('..\\file')` is also accepted, but in linux, only `/` is accepted. This is the reason to check it with both separators.

Reply:
but is `path.sep` `\\` ? Otherwise we're testing just with `\`, are we not? 

I wonder if we could do some logic based on `process.platform` if this is done because of windows. It'd make it easier to follow. Something like: 

(just an idea) 
```javascript
const isDependencyWithSlash = isDependencyWithSeparator(filename, request, '/')
if (process.platform === 'win32') {
  return isDependencyWithSlash || isDependencyWithSeparator(filename, request, path.sep)
}
return isDependencyWithSlash
```

Reply:
Yes, it would be exactly the same, and `isDependencyWithSlash` sounds better than `result` 😄

Reply:
\\ is _not_ valid on Linux though, so this _should not_ check this way on Linux. It should split the checks based on `process.platform`. I would maybe make isDependencyWithSeparator take an array of separators and check every separator valid on the given platform.

Reply:
But it is not checking `\\` in linux. We are only checking `\\` in platform=win32. (before the previous commit, only when the `path.sep` was different to `/`)